v2g-liberty/rootfs/root/appdaemon/apps/v2g_liberty/modbus_evse_client.py

```diff
@@ -1042,90 +1123,121 @@ async def __base_polling(self, kwargs):
             entity_id="sensor.charger_connection_status",
-            state=msg,
+            new_value=msg,
             attributes=keep_alive,
         )
 
     async def __force_get_register(
-        self, register: int, min_value: int, max_value: int
+        self,
+        register: int,
+        min_value_at_forced_get: int,
+        max_value_at_forced_get: int,
+        min_value_after_forced_get: int = None,
+        max_value_after_forced_get: int = None,
     ) -> int:
-        """When a 'realtime' reading from the modbus server is needed that is expected
-        to be between min_value/max_value. We keep trying to get a reading that is within min_value - max_value range.
-
-        Of course there is a timeout. If this is reached we expect the modbus server to have crashed,
-        and we notify the user.
-
+        """
+        When a 'realtime' reading from the modbus server is needed, as opposed to
+        stored value from polling.
+        It is expected to be between min_value_at_forced_get/max_value_at_forced_get.
+        Keep trying to get a reading that is within min_value_at_forced_get - max_value_at_forced_get range or
+        a timeout has been reached.
+        After the timeout the value is checked against other min/max values. This is
+        a wider range.
+        This is aimed at the SoC, this is expected to be between 2 and 97%, but at
+        timeout 1% to 100% is acceptable.
+
+        If the value is not in the wider acceptable range at timeout we assume
+        the modbus server has crashed, and we call __handle_un_recoverable_error.
+
+        :param register: The address to read from
+        :param min_value_at_forced_get: min acceptable value
+        :param max_value_at_forced_get: max acceptable value
+        :param min_value_after_forced_get: min acceptable value after the timeout
+        :param max_value_after_forced_get: max acceptable value after the timeout
+        :return: the read value
         """
         if not self._am_i_active:
-            self.__log(
-                "__force_get_register called while _am_i_active == False. Not blocking."
-            )
+            self.__log("called while _am_i_active == False. Not blocking.")
 
         # Times in seconds
         total_time = 0
-        MAX_TOTAL_TIME = 120
-        DELAY_BETWEEN_READS = 0.25
+        delay_between_reads = 0.25
 
-        # If the real SoC is not available yet, keep trying for max. two minutes
+        # If the real SoC is not available yet, keep trying for
+        # max. self.MAX_CHARGER_ERROR_STATE_DURATION_IN_SECONDS
         while True:
             result = None
             try:
                 # Only one register is read so count = 1, the charger expects slave to be 1.
-                # I hate using the word 'slave', this should be 'server' but pyModbus has not changed this yet
+                # I hate using the word 'slave', this should be 'server' but pyModbus has
```

Comment:
Although I appeciate the sentiment, I hardly think this makes the code more readable.
You may want to consider refraining your personal opinions in code.
I do not expect this to ever change in pymodbus, as this is a simply a part of the modbus nomenclature, that is not going to change.

Reply:
😔